### PR TITLE
Adds tests for UI control classes.

### PR DESCRIPTION
**Pull request by chriscox**

### CODE REVIEW

File: webpack.config.js [hunk 1]
```diff
@@ -60,10 +60,6 @@ module.exports = {
       enforce: 'post'
     }],
   },
-  externals: {
```
Comment (chuga) [inline webpack.config.js:63]: this change seems unrelated, just pointing it out in case you didn't mean to add it here

**Review by chuga — APPROVED** [review]
lgtm

File: webpack.config.js (hunk 1 above)
Comment (chriscox) [reply webpack.config.js:63]: It's actually needed for the tests which require the react dependency.
